builtin/credential/aws/path_login.go

```diff
@@ -208,13 +206,22 @@ func (b *backend) pathLoginUpdate(
 		return nil, err
 	}
 
+	clientNonce := data.Get("nonce").(string)
+	if clientNonce == "" && !storedIdentity.DisallowReauthentication {
```

Comment:
@jefferai Thanks for pointing this out.

If we check for empty nonce later or if we parse the tag earlier, we would have done a lot of processing before input checking.

If storage entry is nil, it indicates first time login. So, I fail if the client nonce is empty. A successful login will properly set the DisallowReauthentication in storage entry, irrespective of whether it came from image or role tag.

Sequent login attempts will land in validateMetadata, and there I have a secondary check for clientNonce being empty, after the disallowReauthentication check. This will ensure that we don't miss out on checking the empty client nonce, all the while ensuring that we are not checking for it too late as well.

Kindly check if this takes care of the problem.